### PR TITLE
Add uefi tests

### PR DESCRIPTION
**Pull request by GanShun**
Signed-off-by: Gan Shun Lim <ganshun@gmail.com>

### CODE REVIEW

File: uefi/uefi_test.go [hunk 1]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+func TestChecksum16(t *testing.T) {
+	var tests = []struct {
+		name string
+		buf  []byte
+		res  uint16
+		msg  string
+	}{
+		{"emptyBuf", emptyBuf, 0, ""},
+		{"sampleBuf", sampleBuf, 0x46, ""},
+		{"overBuf", overBuf, 0x11, ""},
```
Comment (rjoleary) [inline uefi/uefi_test.go]: This seems incorrect. Shouldn't it be 0x200?

File: uefi/uefi_test.go [hunk 2]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+	}{
+		{"emptyBuf", emptyBuf, 0},
+		{"sampleBuf", sampleBuf, 10},
+		{"overBuf", overBuf, 0x1},
+		{"zeroBuf", zeroBuf, 0},
+		{"threeBuf", threeBuf, 9},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (rjoleary) [inline uefi/uefi_test.go]: Remove `t.Parallel()`. It is creating false positives.

File: uefi/uefi_test.go [hunk 3]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+		{"emptyBuf", emptyBuf, 0, ""},
+		{"sampleBuf", sampleBuf, 0x46, ""},
+		{"overBuf", overBuf, 0x11, ""},
+		{"zeroBuf", zeroBuf, 0, ""},
+		{"threeBuf", threeBuf, 0, fmt.Sprintf("byte slice does not have even length, not able to do 16 bit checksum. Length was %v",
+			len(threeBuf))},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (rjoleary) [inline uefi/uefi_test.go]: Remove `t.Parallel()`. It is creating false positives.

File: uefi/uefi_test.go [hunk 4]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+		res  [3]byte
+	}{
+		{"emptySize", 0x0, [3]byte{0, 0, 0}},
+		{"sampleSize", 0xABCDEF, [3]byte{0xEF, 0xCD, 0xAB}},
+		{"max3ByteSize", 0xFFFFFF, [3]byte{0xFF, 0xFF, 0xFF}},
+		{"over3ByteSize", 0x1000000, [3]byte{0xFF, 0xFF, 0xFF}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (rjoleary) [inline uefi/uefi_test.go]: Remove `t.Parallel()`. It is creating false positives.

File: uefi/uefi_test.go [hunk 5]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+		val  uint64
+		arr  [3]byte
+	}{
+		{"emptySize", 0x0, [3]byte{0, 0, 0}},
+		{"sampleSize", 0xABCDEF, [3]byte{0xEF, 0xCD, 0xAB}},
+		{"max3ByteSize", 0xFFFFFF, [3]byte{0xFF, 0xFF, 0xFF}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (rjoleary) [inline uefi/uefi_test.go]: Remove `t.Parallel()`. It is creating false positives.

File: uefi/uefi_test.go [hunk 6]
```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
...
+}
+
+func TestAlign8(t *testing.T) {
+	var tests = []struct {
+		val uint64
+		res uint64
+	}{
+		{0x4, 0x8},
+		{0x5, 0x8},
+		{0x8, 0x8},
```
[the remark this answers is not in the document]
Comment (GanShun) [reply uefi/uefi_test.go]: Done

File: uefi/uefi_test.go (hunk 1 above)
Comment (GanShun) [reply uefi/uefi_test.go]: Fixed

File: uefi/uefi_test.go (hunk 2 above)
Comment (GanShun) [reply uefi/uefi_test.go]: Done

File: uefi/uefi_test.go (hunk 3 above)
Comment (GanShun) [reply uefi/uefi_test.go]: Done

File: uefi/uefi_test.go (hunk 4 above)
Comment (GanShun) [reply uefi/uefi_test.go]: Done

File: uefi/uefi_test.go (hunk 5 above)
Comment (GanShun) [reply uefi/uefi_test.go]: Done

**Pull request merged**